### PR TITLE
Ruby program

### PR DESCRIPTION
**Pull request by Iamyuvraj143**

### CODE REVIEW

File: arrays.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+friends = %w(Shubham Chirag Yuvraj)
+family  = Array[]
```
Comment (kalpana19sharma) [inline arrays.rb:2]:
```suggestion
family  = []
```

Comment (kalpana19sharma) [reply arrays.rb:2]: And actually you don't need it since you are re-initializing this variable in next line

File: data.csv [hunk 2]
```diff
@@ -0,0 +1,6 @@
+id,name
+1,Yuvraj
+2,Singh
+3,Dodiya
+
+4,Yuvi
```
Comment (kalpana19sharma) [inline data.csv]: Add new line

File: double_element_in_array.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Use the map function to double all the elements in the array.
+array = [1,2,3]
+new_array = array.map { |n| n * 2 }
+
+puts  new_array
```
Comment (kalpana19sharma) [inline double_element_in_array.rb]: Add new line

File: double_element_in_array.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Use the map function to double all the elements in the array.
+array = [1,2,3]
```
Comment (kalpana19sharma) [inline double_element_in_array.rb]: space after `,`

File: double_element_in_array2.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#Use the map function to double all the elements in the array.
+# However, handle edge cases (like array can have a character) as well.
+array = [1 , 2 , 3 , "a" , "b"]
```
Comment (kalpana19sharma) [inline double_element_in_array2.rb]:
```suggestion
array = [1, 2, 3, "a", "b"]
```

File: double_element_in_array2.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+new_array = array.map { |n|
+if n.kind_of? Integer
+  n * 2
+end
+ }
```
Comment (kalpana19sharma) [inline double_element_in_array2.rb]: use do..end syntax and single line syntax for conditional statement

File: frequency_of_char.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+#2. Using hash table, print the frequency of occurrence of each character inside an array.
+str = "YHuvrahsnnvdfvvsjvnddsn"
+str =  str.split("")
+counts = {}
+str.group_by(&:itself).each { |k,v| counts[k] = v.length }
```
Comment (kalpana19sharma) [inline frequency_of_char.rb]:
```suggestion
str.group_by(&:itself).each { |k, v| counts[k] = v.length }
```

File: frequency_of_char.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+puts counts
+
+
+# another solution using Enumrable tally
+char = "dfgjdfnsdfgdnv,msdfnw"
+names = names.split("")
+puts names.tally
```
Comment (kalpana19sharma) [inline frequency_of_char.rb]: new line

File: hashes.rb [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+   "Rajastan" => "Rj"
+ }
+
+ puts states
+
+number_in_words = {
+    one: 1 ,
+    two: 2 ,
+    three: 3 ,
+    four: 4 ,
```
Comment (kalpana19sharma) [inline hashes.rb:13]:
```suggestion
    one: 1,
    two: 2,
    three: 3,
    four: 4,
```

File: read_csv.rb [hunk 10]
```diff
@@ -0,0 +1,10 @@
+require 'csv'
+file1 = CSV.read("data.csv")
+
+ puts file1
+
+
+File.open("data.csv" ,"r+") do |file|
+ puts file.read()
+ file.write("\n4,Yuvi")
+end
```
Comment (kalpana19sharma) [inline read_csv.rb]: Fix indentation

**Review by kalpana19sharma — CHANGES_REQUESTED** [review]

File: arrays.rb (hunk 1 above)
Comment (Iamyuvraj143) [reply arrays.rb:2]: Done

File: data.csv (hunk 2 above)
Comment (Iamyuvraj143) [reply data.csv]: Done

File: double_element_in_array.rb (hunk 3 above)
Comment (Iamyuvraj143) [reply double_element_in_array.rb]: Done

File: double_element_in_array.rb (hunk 4 above)
Comment (Iamyuvraj143) [reply double_element_in_array.rb]: Done

File: double_element_in_array2.rb (hunk 5 above)
Comment (Iamyuvraj143) [reply double_element_in_array2.rb]: Done

File: double_element_in_array2.rb (hunk 6 above)
Comment (Iamyuvraj143) [reply double_element_in_array2.rb]: Done

File: frequency_of_char.rb (hunk 7 above)
Comment (Iamyuvraj143) [reply frequency_of_char.rb]: Done

**Review by kalpana19sharma — APPROVED** [review]

**Pull request merged**